tcms/rpc/api/bug.py

```diff
@@ -34,12 +34,16 @@ def details(url, **kwargs):
         :rtype: dict
     """
     result = cache.get(url)
-    if not result:
-        request = kwargs.get(REQUEST_KEY)
-        tracker = tracker_from_url(url, request)
-        result = tracker.details(url)
-        cache.set(url, result)
+    if result:
+        return result
 
+    request = kwargs.get(REQUEST_KEY)
+    tracker = tracker_from_url(url, request)
+    if not tracker:
+        return {}
+
+    result = tracker.details(url)
+    cache.set(url, result)
     return result
```

Comment:
This entire method is missing test coverage so before refactoring it you need to add tests which are expected to continue PASSing even after the refactoring:
https://codecov.io/gh/kiwitcms/Kiwi/src/master/tcms/rpc/api/bug.py